controllers/storagecluster/cephrbdmirrors.go

```diff
@@ -0,0 +1,136 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+
+	ocsv1 "github.com/openshift/ocs-operator/api/v1"
+	"github.com/openshift/ocs-operator/controllers/defaults"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type ocsCephRbdMirrors struct{}
+
+func (r *StorageClusterReconciler) fetchCephRbdMirrorInstance(cephRbdMirror *cephv1.CephRBDMirror) (*cephv1.CephRBDMirror, error) {
+	existing := &cephv1.CephRBDMirror{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: cephRbdMirror.Name, Namespace: cephRbdMirror.Namespace}, existing)
+	return existing, err
+}
+
+func (r *StorageClusterReconciler) deleteCephRbdMirrorInstance(cephRbdMirrors []*cephv1.CephRBDMirror) error {
+	for _, cephRbdMirror := range cephRbdMirrors {
+		existing, err := r.fetchCephRbdMirrorInstance(cephRbdMirror)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return fmt.Errorf("Unable to retrieve CephRbdMirror %v: %v", cephRbdMirror.Name, err)
+			}
+			continue
+		}
+		if cephRbdMirror.GetDeletionTimestamp().IsZero() {
+			r.Log.Info("Deleting CephRbdMirror.", "CephRbdMirror", klog.KRef(cephRbdMirror.Namespace, cephRbdMirror.Name))
+			err := r.Client.Delete(context.TODO(), existing)
+			if err != nil {
+				r.Log.Info("Retrying deletion of CephRbdMirror.", "CephRbdMirror", klog.KRef(existing.Namespace, existing.Name))
+				err := r.Client.Delete(context.TODO(), existing)
+				if err != nil {
+					r.Log.Error(err, "Failed to delete CephRbdMirror.", "CephRbdMirror", klog.KRef(existing.Namespace, existing.Name))
```

Comment:
No need to retry. Just return the error and it will requeue.

Reply:
ack